spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -1527,15 +1527,15 @@ def process_started(self, client):
 
         if self.main.variableexplorer is not None:
             self.main.variableexplorer.add_shellwidget(client.shellwidget)
-        if self.main.plots is not None:
-            self.main.plots.add_shellwidget(client.shellwidget)
+
+        self.sig_shellwidget_process_started.emit(client.shellwidget)

```

Comment:
That is not true. Plot uses the `shellwidget` 

https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/plots/plugin.py#L91

And having some signals pass an object and some others pass an ID and have the same name is not very consistent. 

So the name of the signals need to change, but the plot widget still needs the shellwidget, because that is how the plugin works.

Reply:
Ok, I'm sorry for my mistake, you were right. Then It's fine to leave things as you implemented them. Thanks for the clarification.